src/background/uri-info.js

```diff
@@ -31,43 +30,51 @@ function shouldQueryUri(uri) {
     return false;
   }
 
-  // Make sure `uri` does not match ANY item in the blocklist
-  return BLOCKLIST.every(
-    blockedItem => url[blockedItem.urlProperty] !== blockedItem.value
-  );
+  if (!ALLOWED_PROTOCOLS.has(url.protocol)) {
+    return false;
+  }
+
+  if (BLOCK_HOSTNAMES.has(url.hostname)) {
+    return false;
+  }
+
+  return true;
 }
 
 /**
  * Queries the Hypothesis service that provides
  * statistics about the annotations for a given URL.
+ *
+ * @return {Promise<number>}
+ * @throws Will throw a variety of errors: network, json parsing, or wrong format errors.
  */
-function query(uri) {
-  return fetch(settings.apiUrl + '/badge?uri=' + encodeUriQuery(uri), {
-    credentials: 'include',
-  }).then(res => res.json());
+async function query(uri) {
+  const response = await fetch(
+    settings.apiUrl + '/badge?uri=' + encodeUriQuery(uri),
+    {
+      credentials: 'include',
+    }
+  );
+
+  const data = await response.json();
+
+  if (data && typeof data.total === 'number') {
+    return data.total;
+  }
+
+  throw new Error('Badge response has wrong format');
```

Comment:
If parsing fails it doesn't necessarily mean that the format is "wrong". It might be that we change the API in future and an old extension is installed somewhere which is unable to parse it. "Unable to parse badge response" would be more accurate.